### PR TITLE
fix(utxo-lib): use PsbtInput instead of UtxoPsbt

### PR DESCRIPTION
**Pull request by saravanan7mani**
Currently, many psbt PsbtInput related util funtions depend
on UtxoPsbt and input index number as inputs.

This unnecessarily causes some code complexity.

This PR replaces them directly with PsbtInput as input.

Ticket: BG-78639

BREAKING CHANGE: functions signature is changed

### CODE REVIEW

File: modules/utxo-lib/src/bitgo/PsbtUtil.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { PsbtInput } from 'bip174/src/lib/interfaces';
+
+/**
+ * alias for PsbtInput type to avoid direct bip174 library dependency by users of the util functions
+ */
+export type PsbtInputType = PsbtInput;
+
+/**
+ * To search any data from psbt proprietary key value against keydata.
+ * Default identifierEncoding is utf-8 for identifier.
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/PsbtUtil.ts:17]:
```suggestion
 * Search any data from psbt proprietary key value against keydata.
 * Default identifierEncoding is utf-8 for identifier.
```

File: modules/utxo-lib/src/bitgo/PsbtUtil.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import { decodeProprietaryKey } from 'bip174/src/lib/proprietaryKeyVal';
+import {
+  ProprietaryKeySearch,
+  ProprietaryKeySubtype,
+  ProprietaryKeyValue,
+  PSBT_PROPRIETARY_IDENTIFIER,
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/PsbtUtil.ts:6]: You can probably move these into this file

File: modules/utxo-lib/src/bitgo/PsbtUtil.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { decodeProprietaryKey } from 'bip174/src/lib/proprietaryKeyVal';
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/PsbtUtil.ts:1]: `PsbtUtil.ts` is a bit generic - let's come up with some better names

File: modules/utxo-lib/src/bitgo/PsbtUtil.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import {
+  ProprietaryKeySearch,
+  ProprietaryKeySubtype,
+  ProprietaryKeyValue,
+  PSBT_PROPRIETARY_IDENTIFIER,
+} from './UtxoPsbt';
+import { PsbtInput } from 'bip174/src/lib/interfaces';
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/PsbtUtil.ts:8]: external imports should precede internal ones

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
nice cleanup!

File: modules/utxo-lib/src/bitgo/PsbtUtil.ts (hunk 1 above)
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/PsbtUtil.ts]: Done

File: modules/utxo-lib/src/bitgo/PsbtUtil.ts (hunk 2 above)
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/PsbtUtil.ts]: Done

File: modules/utxo-lib/src/bitgo/PsbtUtil.ts (hunk 4 above)
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/PsbtUtil.ts:2]: Done

**Review by OttoAllmendinger — APPROVED** [review]

**Review by aarvay — APPROVED** [review]

**Review by Phani024 — APPROVED** [review]

**Pull request merged**